sdk/state/update.go

```diff
@@ -0,0 +1,143 @@
+package state
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/stellar/experimental-payment-channels/sdk/txbuild"
+	"github.com/stellar/go/xdr"
+)
+
+type PaymentProposal struct {
+	IterationNumber            int64
+	ObservationPeriodTime      time.Duration
+	ObservationPeriodLedgerGap int64
+	AmountToInitiator          int64
+	AmountToResponder          int64
+	CloseSignatures            []xdr.DecoratedSignature
+	DeclarationSignatures      []xdr.DecoratedSignature
+}
+
+// TODO - validate inputs? (eg. no negative amounts)
+// TODO - payments to be in Amount struct
+// initiator will only call this
+func (c *Channel) NewPaymentProposal(payToInitiator int64, payToResponder int64) (*PaymentProposal, error) {
```

Comment:
I think you can reduce this interface down to a single amount, the amount the caller wants to send.

```suggestion
func (c *Channel) NewPaymentProposal(amount int64) (*PaymentProposal, error) {
```


Inside the function we can do the following to update an initiator/responder balance, which I see is what's in this PR. The rest of your logic would remain the same inside the function.

```go
if c.initiator {
    c.balance += amount
} else {
    c.balance -= amount
}
```

We could also support a local/remote balance like so, but I don't yet have an opinion on which would have better properties:

```go
c.balance += amount
amountToInitiator := c.initiatorAmountOwed()
amountToResponder := c.responderAmountOwed()
```

We would have functions like `initiatorAmountOwed()` that convert the local representation into the initiator's perspective, and similar for responder.

Reply:
reduced to a single amount input: https://github.com/stellar/experimental-payment-channels/pull/55/commits/c84ca0309abd42cd75d04d1f491b82f8c89f1bac